### ISSUE TITLE
Refactor the code

### ISSUE THREAD

**Issue by nikhilrayaprolu** [fossasia/susper.com#32]
It looks likes the susper.com 's code is not organized , many script files are  written inside the index.html without  referencing them from external files, including some css.
In css absolute positions are used which could be replaced with relative positions ( using bootstrap replacing custom css could also make the site responsive).
Refactoring the code makes us easier to maintain susper in future and even comfortable to extend it, and easier for new contributors to start with.

**Comment by mariobehling** [comment]
Would be great, if you work on it.

**Comment by Marauderer97** [comment]
@nikhilrayaprolu @mariobehling 
I would like to help with this, @nikhilrayaprolu where do I start?

**Comment by nikhilrayaprolu** [comment]
@Marauderer97 Yeah! go ahead! I would suggest starting with separating java script and css first, and then changing the css to make it more relative, using bootstap where you can.

**Comment by mariobehling** [comment]
@nikhilrayaprolu @Marauderer97 excellent initiative. Please feel free to add subissues. We definitely need this.

**Comment by Marauderer97** [comment]
Hello @mariobehling  I have refactored the index page of susper, but something keeps going wrong when I try to seperate the javascript from the html in yacysearch, I am working on it. Do you want me to make a PR?

**Comment by mariobehling** [comment]
@Marauderer97 Take your time to get it right and make a PR then. Thank you.

**Comment by nikhilrayaprolu** [comment]
@mariobehling why not rewrite susper.com in angular2 , or react or using some frontend library.
this makes us easier to maintain, clean ,write tests and easier to extend functionalities.
even integrating with travis-ci or others platforms get better

**Comment by mariobehling** [comment]
@nikhilrayaprolu Angular2 sounds like a good idea. Would you be willing to take this on?

**Comment by Marauderer97** [comment]
@mariobehling Yes I am sure he would. @nikhilrayaprolu and I have discussed this, and we both are willing to take it on :)

**Comment by mariobehling** [comment]
@Marauderer97 @nikhilrayaprolu Ok, sounds great. If you are up for it, when do you think you could have a prototype in Angular2?

**Comment by Marauderer97** [comment]
@mariobehling we have registrations for the spring semester in college tomorrow, so we will come with a prototype in Wednesday? :)

**Comment by mariobehling** [comment]
Ok, sounds great. Thank you.

**Comment by nikhilrayaprolu** [comment]
@mariobehling so far we have decided that we could have atleast two main components one for  the initial search page and the other for search results. we would like to  use angular-cli for building the project. can you give us anymore suggestions/ideas regarding this?
we are planning on creating a new branch for angular2 within this repo and commit our work to it.should we proceed?

**Comment by mariobehling** [comment]
@nikhilrayaprolu Sounds good. Please open a parent issue for the transition to Angular2 and open subissues for each area you are working on. It would be great to include others in this process and following the issue workflow could make it easier for others.

**Comment by nikhilrayaprolu** [comment]
@mariobehling can you create a new branch in this repo for angular2 to which we could commit all our work.

**Comment by Rohanhacker** [comment]
why not react ?

**Comment by nikhilrayaprolu** [comment]
@Rohanhacker thankyou for showing interest on this discussion.React is a very good library ("I don't call it a framework") . but if we consider angular2 ,we could see many new better features in it, and a full support as a framework (where as react is just a library and we need to extend it with other libraries for some extra functionalities). and learning angular2 is not hard for you ,as you are from react background and angular2 is syntactically similar to react (component based and one way bindings).
infact the extra features like angualar-cli (inspired from ember) ,lazy loading ,dynamic loading,change detection,Ahead of time(AOT) will make angular2 completely different.
look at ngrx , a redux library for angular2(ofcourse its an inspiration took from react) is lot more easier to implement compared to react-redux.
in my opinion angular2 is a way preferable as it has inherited features from react ,ember and its own parent version angular v1.0 and also adding some new features along the way.
Even the community has decided to change the way of releasing to SEMVER with 
    patch releases every week,
    3 monthly minor release after each major release and
    a major release with easy-to-migrate-over breaking changes every 6 months.
which makes us updated with new features of the javascript community all the time.

**Comment by Rohanhacker** [comment]
@nikhilrayaprolu I think react is more popular than angular 2 so we'll be having more people already knowing react than angular 2, also beginners can face difficulty with observes and stuff.

**Comment by mariobehling** [comment]
@Rohanhacker Both are very popular and there is a huge crowd of people who could support either. I think the approaches are just different. If each of you prefer a different approach, you could build an app in Angular and React. We can create another reop, if you like.

**Comment by Rohanhacker** [comment]
@nikhilrayaprolu are you working on angular2 prototype ?

**Comment by nikhilrayaprolu** [comment]
yep @Rohanhacker ,I have started working on it.

**Comment by Rohanhacker** [comment]
@nikhilrayaprolu It would be great of you can push your work on a separate branch and create a more detailed roadmap.

**Comment by nikhilrayaprolu** [comment]
On 18 January 2017 at 17:26, Rohan Malhotra <notifications@github.com>
wrote:

> @nikhilrayaprolu <https://github.com/nikhilrayaprolu> It would be great
> of you can push your work on a separate branch and create a more detailed
> roadmap.
>
@Rohanhacker
I have already made a prototype and detailed time line for the road map .
you could check it at the parent issue raised at
https://github.com/fossasia/susper.com/issues/43 . also the work related to
angular is being done at a new branch named 'angular' . take a look at it
,and any inputs are appreciated.
